### PR TITLE
[JS] Add csrf middleware documentation

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates to: https://github.com/SAP/cloud-sdk-backlog/issues/1043
Relates to: https://github.com/SAP/cloud-sdk-backlog/issues/1049

## What Has Changed?

Add the CSRF middleware to the guide.

### CODE REVIEW

File: docs-js/features/connectivity/csrf.mdx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+---
+id: csrf
+title: Csrf Fetching
```
Comment (deekshas8) [inline docs-js/features/connectivity/csrf.mdx:3]: [pp] CSRF Token Fetching (same for sidebar label)

File: docs-js/features/connectivity/csrf.mdx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+## Default Behaviour
+
+The SAP Cloud SDK fetches a CSRF token for non `GET` requests per default.
+This is done no for all clients the SAP Cloud SDK provides:
+
+- [generic http-client](./http-client#csrf-token-fetching)
+- [v2 OData clients](../odata/v2-client#handling-of-cross-site-request-forgery-tokens)
+- [v4 OData clients](../odata/v4-client#handling-of-cross-site-request-forgery-tokens)
+- [OpenApi clients](../openapi/execute-request#handling-of-cross-site-request-forgery-tokens).
```
Comment (deekshas8) [inline docs-js/features/connectivity/csrf.mdx:28]:
```suggestion
- [OpenApi clients](../openapi/execute-request#handling-of-cross-site-request-forgery-tokens)
```

File: docs-js/features/connectivity/csrf.mdx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+If the target system does not require a token you can disable the fetching as discussed in the sections:
+
+- [disable for http-client](./http-client#csrf-token-fetching)
+- [disable for v2 OData clients](../odata/v2-client#hskip-csrf-token-handling)
+- [disable for v4 OData clients](../odata/v4-client#skip-csrf-token-handling)
+- [disable OpenApi clients](../openapi/execute-request#skip-csrf-token-handling).
```
Comment (deekshas8) [inline docs-js/features/connectivity/csrf.mdx:35]:
```suggestion
- [disable OpenApi clients](../openapi/execute-request#skip-csrf-token-handling)
```

File: docs-js/features/connectivity/csrf.mdx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+In order to retrieve a token the SAP cloud SDK does the following steps:
+
+Makes a request using the resource URL adding a `/` in the end:
+
+```
+Request:             HEAD /reource/path/
```
Comment (deekshas8) [inline docs-js/features/connectivity/csrf.mdx:42]:
```suggestion
Request:             HEAD /resource/path/
```

File: docs-js/features/connectivity/csrf.mdx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+Request Header:      X-CSRF-Token=fetch
+```
+
+If this returns a token execute the request with the token.
+If not, make a second call without the `/` is tried:
+
+```
+Request:             HEAD /reource/path
```
Comment (deekshas8) [inline docs-js/features/connectivity/csrf.mdx:50]:
```suggestion
Request:             HEAD /resource/path
```

File: docs-js/features/connectivity/csrf.mdx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+## Changing the Default
+
+If the way the SAP Cloud SDK fetches the CSRF token does not fit your needs you can replace the default using a middleware.
+The SAP Cloud SDK offers a default implementation which allows minor adjustments as described here:
+
+- [disable for http-client](./http-client#csrf-token-fetching)
+- [disable for v2 OData clients](../odata/v2-client#custom-csrf-token-handling)
+- [disable for v4 OData clients](../odata/v4-client#custom-csrf-token-handling)
+- [disable OpenApi clients](../openapi/execute-request#custom-csrf-token-handling).
```
Comment (deekshas8) [inline docs-js/features/connectivity/csrf.mdx:79]:
```suggestion
- [disable for OpenApi clients](../openapi/execute-request#custom-csrf-token-handling)
```

File: docs-js/features/connectivity/csrf.mdx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+- [OpenApi clients](../openapi/execute-request#handling-of-cross-site-request-forgery-tokens).
+
+If the target system does not require a token you can disable the fetching as discussed in the sections:
+
+- [disable for http-client](./http-client#csrf-token-fetching)
+- [disable for v2 OData clients](../odata/v2-client#hskip-csrf-token-handling)
+- [disable for v4 OData clients](../odata/v4-client#skip-csrf-token-handling)
+- [disable OpenApi clients](../openapi/execute-request#skip-csrf-token-handling).
+
+In order to retrieve a token the SAP cloud SDK does the following steps:
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply docs-js/features/connectivity/csrf.mdx:37]: adjusted

**Review by deekshas8 — APPROVED** [review]

**Pull request merged**